renku/core/dataset/dataset.py

```diff
@@ -1168,3 +1170,90 @@ def should_include(filepath: Path) -> bool:
         raise errors.ParameterError(f"These datasets don't exist: {unused_names_str}")
 
     return sorted(records, key=lambda r: r.date_added)
+
+
+def pull_external_data(name: str, location: Optional[Path] = None) -> None:
+    """Pull/copy data for an external storage to a dataset's data directory or a specified location.
+
+    Args:
+        name(str): Name of the dataset
+        location(Optional[Path]): A directory to copy data to (Default value = None).
+    """
+    client = get_client()
+    datasets_provenance = DatasetsProvenance()
+
+    dataset = datasets_provenance.get_by_name(name=name, strict=True)
+    assert dataset  # NOTE: To make mypy happy
```

Comment:
e.g. https://github.com/SwissDataScienceCenter/renku-python/pull/3071/files#diff-44504c88307d8f6a0c6e125d4d1de71e45b73c90fb6109dc95c27c62194cc9c9R60 works. It's annoying that we need to add `*,` so it's a keyword arg, because we can't define two default values in the overloads, otherwise mypy complains. But it works pretty nicely otherwise.

Reply:
This won't work in Python 3.7 because `Literal` was added in 3.8. I'm not sure why we changed the base Python for testing from 3.7 to 3.9.